### PR TITLE
Add PHP 8.0 support

### LINKED ISSUE #3: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by Slamdunk**
Close https://github.com/laminas/laminas-cache-storage-adapter-memcache/issues/3

### CODE REVIEW

**Comment by boesing** [comment]
@Slamdunk Can we create a dedicated PR for the coding standard?
I'd like to have a clean diff to see what changes were needed for PHP 8.0.

The idea behind this is having PRs to link to when others face the same issues while migrating to PHP 8.0.
This gives a good overview on which pitfalls may be occur when migrating to PHP 8.0.

File: test/integration/Psr/SimpleCache/MemcacheIntegrationTest.php [hunk 1]
```diff
@@ -67,6 +59,8 @@ public function createSimpleCache(): SimpleCacheDecorator
             $options['servers'] = [[$host]];
         }
 
+        $this->skippedTests['testBasicUsageWithLongKey'] = 'SimpleCacheDecorator requires keys to be <= 64 chars';
```
Comment (boesing) [inline test/integration/Psr/SimpleCache/MemcacheIntegrationTest.php:62]: Why is there an integration test for long keys if they're not supported by the PSR standard?
Or does memcache limit the cache key to `<= 64 chars`?
If not, we should not skip the test but fix the problem which is causing the issue?

**Review by boesing — CHANGES_REQUESTED** [review]

File: test/integration/Psr/SimpleCache/MemcacheIntegrationTest.php (hunk 1 above)
Comment (Slamdunk) [reply test/integration/Psr/SimpleCache/MemcacheIntegrationTest.php:62]: Nothing to do with `memcache`, here the sources:

1. https://github.com/php-cache/integration-tests/pull/86#discussion_r621180635
2. https://github.com/laminas/laminas-cache/commit/5e10ef6222ff342e651387630ea1637aa7c68e29

**Comment by Slamdunk** [comment]
> @Slamdunk Can we create a dedicated PR for the coding standard?
> I'd like to have a clean diff to see what changes were needed for PHP 8.0.
> 
> The idea behind this is having PRs to link to when others face the same issues while migrating to PHP 8.0.
> This gives a good overview on which pitfalls may be occur when migrating to PHP 8.0.

You are right and your request is reasonable and agreeable...
I just tried to improve the code all at once to reduce the work for both; given the little size of the codebase and the diff, I hope you can accept it as is so I can focus my attention on the other satellites, were I'm push everything all in one PR, but on different commits.

**Comment by boesing** [comment]
> [...] given the little size of the codebase and the diff, I hope you can accept it as is [...]

Yah, as this is only one file I have to verify no BC breaks are coming in, this is an option.

**Review by boesing — APPROVED** [review]
LGTM

File: test/integration/Psr/SimpleCache/MemcacheIntegrationTest.php (hunk 1 above)
Comment (boesing) [reply test/integration/Psr/SimpleCache/MemcacheIntegrationTest.php:62]: Hmm, I am torn.
The PSR states that every PSR-16 implementing cache **must** support 64 chars. It **can** support more than that.
So imho, https://github.com/laminas/laminas-cache/commit/5e10ef6222ff342e651387630ea1637aa7c68e29 was based on a wrong assumption on how PSR-16 works.

I am fine with this for the 1.x versions.
Imho, having a new required option for the `AdapterOption` for v3 would help us optimize the decorators.

So I'll create a new Issue for that.

Comment (boesing) [reply test/integration/Psr/SimpleCache/MemcacheIntegrationTest.php:62]: https://github.com/laminas/laminas-cache/issues/91

**Pull request merged**